db/pebble/db.go

```diff
@@ -64,16 +64,18 @@ func (db *DB) Close() error {
 }
 
 // View : see db.DB.View
-func (d *DB) View(fn func(txn db.Transaction) error) (err error) {
+func (d *DB) View(fn func(txn db.Transaction) error) error {
 	txn := d.NewTransaction(false)
+	var err error
 	defer db.CloseAndWrapOnError(txn.Discard, &err)
```

Comment:
The behaviour of `db.CloseAndWrapOnError` has changed because the error returned by `fn(txn)` on line 72 will not be set to the error declared on line 69. Thus the error passed to `db.CloseAndWrapOnError` will always be nil.

This is the case in most places where `db.CloseAndWrapOnError` is being used. Either we change how we wrap errors or we set `err` correctly before returning. `db.CloseAndWrapOnError` relied on the named return parameter to work correctly. Since we are no longer using that `db.CloseAndWrapOnError` no longer works as expected.